lib/swagger-catalog.js

```diff
@@ -25,57 +33,134 @@ function getServiceSummary(service) {
+  return new Promise((resolve) => {
+    const openApi = {
+      openapi: "3.0.0",
+      info: {
+        title:  plugin.identifier,
+        description: plugin.descriptionDefault,
+        version: plugin.version || "0.0.1"
       },
-      put: {
-        responses: {
-          200: {
-            description: "service call succeeded"
+      basePath: zLuxUrl.makePluginURL(productCode, plugin.identifier) + "/services",
+      host: os.hostname(),
+      schemes: getSchemesFromContext(nodeContext),
+      paths: {}
+    };
+    getSwaggerDocs(plugin, productCode, nodeContext).then((swaggerDocs) => {
+      if (swaggerDocs.length > 0) {
+        // if there are swagger docs that were read create the paths from those 
+        swaggerDocs.forEach((service) => {
+          var tempPaths = service.serviceDoc.paths;
+          var tempVersion = service.serviceDoc.info.version;
+          for (key in tempPaths) {
+            openApi.paths["/" + service.serviceName + "/" + tempVersion + key] = tempPaths[key];
           }
+        });
+      } else {
+        // if there are no swagger docs that were read use this placeholder setup
+        for (const service of plugin.dataServices || []) {
+          //FIXME templates with an asterisk are not supported by open api
+          //TODO we can actually somewhat inspect Express routers
+          openApi.paths[zLuxUrl.makeServiceSubURL(service) + '/*'] = {
+            summary: getServiceSummary(service),
+            get: {
+              responses: {
+                200: {
+                  description: "service call succeeded"
+                }
+              }
+            },
+            post: {
+              responses: {
+                200: {
+                  description: "service call succeeded"
+                }
+              }
+            },
+            put: {
+              responses: {
+                200: {
+                  description: "service call succeeded"
+                }
+              }
+            },
+            delete: {
+              responses: {
+                200: {
+                  description: "service call succeeded"
+                }
+              }
+            }
+          };
         }
-      },
-      delete: {
-        responses: {
-          200: {
-            description: "service call succeeded"
-          }
+      }
+      // pass everything back at once the plugin catalog with all services together
+      // swaggerdocs is the list of all the docs for each service
+      var allDocumentation = {
+        pluginCatalog: openApi,
+        serviceDocs: swaggerDocs
+      }
+      resolve(allDocumentation)
+    });
+  })
+}
+
+async function getSwaggerDocs(plugin, productCode, nodeContext) {
+  var allServiceDocs = [];
+  if (plugin.dataServices){
+    for (let i = 0; i < plugin.dataServices.length; i++) {
+      let service = plugin.dataServices[i]
+      if (service.swaggerdoc) {
```

Comment:
This complicates things. It's easier to enforce a naming convention: in /doc/swagger a file with the servicename + .json or .yaml should be found if present. If an import dataservice, routers should be attached appropriately later and file reading skipped.